packages/fxa-settings/src/lib/file-utils-legal.tsx

```
@@ -2,74 +2,42 @@
-
-  try {
-    const response = await fetch(`${LEGAL_DOCS_PATH}/${locale}/${file}.md`);
-    if (response.ok) {
-      const resText = await response.text();
-      if (resText.includes('<!DOCTYPE html>')) {
-        return { error };
-      }
-      return { markdown: resText };
-    } else {
-      throw Error(response.statusText);
-    }
-  } catch (e) {
-    sentryMetrics.captureException(e);
-
-    // TODO: If the first preferred language can't be loaded, try
-    // the next 2? and then fallback to English + clean this up
-    if (locale !== 'en') {
-      try {
-        const response = await fetch(`${LEGAL_DOCS_PATH}/en/${file}.md`);
-        if (response.ok) {
-          const resText = await response.text();
-          if (resText.includes('<!DOCTYPE html>')) {
-            return { error };
-          }
-          return { markdown: resText };
-        }
-      } catch (e) {
-        sentryMetrics.captureException(e);
-        return { error };
-      }
-    }
-    return { error };
+export const fetchLegalMd = async (
+  apolloClient: ApolloClient<object> | undefined,
+  locale: string,
+  file: string
+): Promise<{
+  markdown?: string;
+  error?: string;
+}> => {
+  const error = `Something went wrong. Try again later.`;
+
+  if (apolloClient == null) {
+    console.error('No apolloClient provided.');
+    return {
+      error,
+    };
   }
-};
 
-export const fetchLegalMd = async (
-  acceptLanguages: readonly string[],
-  localeParam: string | undefined,
-  file: LegalDocFile
-) => {
-  let availableLocales;
   try {
-    const response = await fetch(`${LEGAL_DOCS_PATH}/${file}_locales.json`);
-    availableLocales = await response.json();
-  } catch (e) {
-    // report to Sentry and allow default locales to be loaded
-    sentryMetrics.captureException(e);
+    const result = await apolloClient?.query({
+      query: GET_LEGAL_DOC,
+      variables: { input: { locale, file } },
+    });
+    return {
+      markdown: result?.data?.markdown,
```

re: `result?.data?.markdown`,
Should this this throw an error if we're getting back `null`|`undefined` results? Or is this fine if it returns `undefined` for `markdown`?

---

We should throw an error.